### PR TITLE
Updates for install_pinned

### PR DESCRIPTION
**Pull request by bmerry**
- Remove redundant bits from Jenkinsfile
- Remove unneeded recursive requirements

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,60 +1,24 @@
...
-python-dateutil           # via pandas, botocore, bokeh
-python-lzf                # via katsdptelstate
-pytools                   # via pycuda
-pytz                      # via pandas
-pyyaml                    # via bokeh
-rdbtools                  # via katsdptelstate
-redis                     # via katsdptelstate
-requests                  # via katdal, katsdpmodels
 scikit-cuda
 scipy                     # via katdal (its fallback interpolation behaves differently)
```
Comment (ludwigschwardt) [inline requirements.txt:21]: This is not the reason anymore (`katdal` has ditched `scipy` and `scikit.fitting` in ska-sa/katdal#292 in one of the last office-based acts). I see that `clean` uses `scipy.stats`, so it is a direct dependency.

File: requirements.txt [hunk 2]
```diff
@@ -1,60 +1,24 @@
...
-chardet                   # via requests
-cityhash                  # via katdal
 contextvars==2.4
-cycler                    # via matplotlib
-dask                      # via katdal
-decorator                 # via pycuda, katsdpsigproc
-docutils                  # via botocore
-ephem                     # via pyephem
-future                    # via katdal, katpoint
-h5py                      # via katdal, katsdpmodels
```
Comment (ludwigschwardt) [inline requirements.txt:18]: `preprocess` and `primary_beam` uses `h5py` directly.

**Review by ludwigschwardt — CHANGES_REQUESTED** [review]
There is also a `scipy` in `tests_require` that is already in `install_requires`. And should `katsdpmodels` in requirements.txt be `katsdpmodels[requests]` instead?

File: requirements.txt (hunk 1 above)
Comment (bmerry) [reply requirements.txt]: Thanks, it was on my TODO list to check if it was still needed after doing this PR, so you've saved me the bother of doing the research :-)

File: requirements.txt (hunk 2 above)
Comment (bmerry) [reply requirements.txt:18]: Thanks, fixed!

File: requirements.txt [hunk 3]
```diff
@@ -1,60 +1,24 @@
+# TODO: switch from -d to -c and unpin jinja2, bokeh once katsdpdockerbase
+# updated.
+-d https://raw.githubusercontent.com/ska-sa/katsdpdockerbase/master/docker-base-build/base-requirements.txt
+
 ansicolors
-appdirs                   # via katsdpsigproc
-argparse                  # via python-casacore
 astropy
 bokeh==2.0.2              # katsdpdockerbase currently provides only 1.x
 botocore                  # for S3 authentication in katdal
```
[the remark this answers is not in the document]
Comment (bmerry) [reply requirements.txt]: Good idea, done. Similarly using katsdpsigproc[CUDA] now.

**Review by ludwigschwardt — APPROVED** [review]

**Pull request merged**